smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/endpointsV2/EndpointsV2Generator.java

```diff
@@ -73,7 +73,7 @@ private void generateEndpointParameters() {
                     new RuleSetParameterFinder(service).getClientContextParams();
 
                 clientContextParams.forEach((k, v) -> {
-                    writer.write("$L: $L,", k, v);
+                    writer.write("$L?: $L,", k, v);
```

Comment:
fix: making the clientContextParams optional at the service client constructor level